evm-adapters/src/call_tracing.rs

```diff
@@ -333,30 +356,26 @@ pub struct CallTraceNode {
 
 impl CallTraceNode {
     /// Prints a log at a particular index, optionally decoding if abi is provided
-    pub fn print_log(&self, index: usize, abi: Option<&Abi>, left: &str) {
+    pub fn print_log(&self, index: usize, events: &BTreeMap<H256, Event>, left: &str) {
         let log = &self.logs[index];
         let right = "  ├─ ";
-        if let Some(abi) = abi {
-            for (event_name, overloaded_events) in abi.events.iter() {
-                for event in overloaded_events.iter() {
-                    if event.signature() == log.topics[0] {
-                        let params = event.parse_log(log.clone()).expect("Bad event").params;
-                        let strings = params
-                            .into_iter()
-                            .map(|param| format!("{}: {}", param.name, format_token(&param.value)))
-                            .collect::<Vec<String>>()
-                            .join(", ");
-                        println!(
-                            "{}emit {}({})",
-                            left.replace("├─", "│") + right,
-                            Colour::Cyan.paint(event_name),
-                            strings
-                        );
-                        return
-                    }
-                }
-            }
+
+        if let Some(event) = events.get(&log.topics[0]) {
```

Comment:
oops nice optimization, iterating over the entire map was a mistake, I should have caught it in review